python/pyspark/sql/dataframe.py

```diff
@@ -482,10 +482,22 @@ def show(self, n=20, truncate=True, vertical=False):
          age  | 5
          name | Bob
         """
+
+        if not isinstance(n, int) or isinstance(n, bool):
```

Comment:
why do we need to check if `n` is `bool`? bool inherits `int` anyway so it's no-op to remove

Reply:
right, so if we pass `n=False` then ` not isinstance(n, int)` is `False` and with this alone the error would not be raised. We need to have an explicit check to reject the `bool`-typed `n`

Reply:
Is this trying to catch the case where a user calls `.show(False)` for example, meaning `.show(truncate=False)`? I get it. While I think we could equally check so many other methods like this, maybe this is worth checking more thoroughly here.

Reply:
Maybe we can come up with a generic mechanism  to tackle  non-existing method exceptions from`ReflectionEngine.getMethod`